kernel/src/mem.rs

```diff
@@ -18,15 +18,15 @@ pub struct Shared;
 /// Base type for an AppSlice that holds the raw pointer to the memory region
 /// the app shared with the kernel.
 pub struct AppPtr<L, T> {
-    ptr: Unique<T>,
+    ptr: NonNull<T>,
     process: AppId,
     _phantom: PhantomData<L>,
 }
 
 impl<L, T> AppPtr<L, T> {
     unsafe fn new(ptr: *mut T, appid: AppId) -> AppPtr<L, T> {
         AppPtr {
-            ptr: Unique::new_unchecked(ptr),
+            ptr: NonNull::new_unchecked(ptr),
```

Comment:
This `unsafe` isn't necessary. Can we pass a `NonNull<T>` as argument instead of a `*mut T`, and build the `NonNull<T>` up the call stack where a null check happens? I see this called from `AppSlice::new`, itself coming from the `allow()` function in `kernel/src/process.rs` where a null pointer check happens.

Reply:
Resolved by bccb481